src/services/firebase.js

```diff
@@ -22,11 +22,48 @@ import '@react-native-firebase/crashlytics';
 import '@react-native-firebase/iid';
 import '@react-native-firebase/analytics';
 import '@react-native-firebase/messaging';
-import '@react-native-firebase/remote-config';
+
+import remoteConfig from '@react-native-firebase/remote-config';
+import { INITIAL_FEATURE_FLAGS } from 'constants/featureFlagsConstants';
+import { log } from 'utils/logger';
 
 
 export const firebaseAnalytics = firebase.analytics();
 export const firebaseIid = firebase.iid();
 export const firebaseCrashlytics = firebase.crashlytics();
 export const firebaseMessaging = firebase.messaging();
 export const firebaseRemoteConfig = firebase.remoteConfig();
+
+
+export const initFirebase = async () => {
+  /**
+   * First, we need to set the defaults for Remote Config.
+   * This makes the default values immediately available
+   * on app load and can be used.
+   *
+   * @url https://rnfirebase.io/reference/remote-config#setDefaults
+   */
+
+  try {
+    remoteConfig().setDefaults(INITIAL_FEATURE_FLAGS);
+    log.info('Firebase Config: Defaults loaded and available.');
+  } catch (e) {
+    log.error('Firebase Config: An error occurred loading defaults:', e);
+  }
+
+  /**
+   * Secondly, we need to activate any remotely fetched values
+   * if they exist at all. The values that have been fetched
+   * and activated override the default values above (see @url
+   * above).
+   *
+   * @url https://rnfirebase.io/reference/remote-config#activate
+   */
+
+  try {
+    const r = remoteConfig().activate();
+    log.info('Firebase Config: Activation result was:', r);
+  } catch (e) {
+    log.error('Firebase Config: An error occurred while activating:', e);
+  }
+};
```

Comment:
Are the try / catches necessary here? Just wanted to explain why i keep pointing this out... from what i recall, .activate is a promise (just checked - it is: https://rnfirebase.io/reference/remote-config#activate) so we should be able to .catch this to handle our errors, instead of try / catch.

To me personally... try / catch is the "nuclear" option - like the last resort (and also the best solution) that we absolutely know that we cannot handle the error using built-in methods. JSON.parse is a good example when it errors, you need to use try / catch. The above isn't "wrong" per-se ... just feels a little bit overkill when we _could_ use `.promise`...

Reply:
Same here - reverted it back :)